src/ucp/core/ucp_worker.c

```diff
@@ -1499,28 +1498,27 @@ ucs_status_t ucp_worker_create(ucp_context_h context,
 
     if (params->field_mask & UCP_WORKER_PARAM_FIELD_THREAD_MODE) {
 #if !ENABLE_MT
-        thread_mode = UCS_THREAD_MODE_SINGLE;
+        uct_thread_mode = UCS_THREAD_MODE_SINGLE;
         if (params->thread_mode != UCS_THREAD_MODE_SINGLE) {
             ucs_debug("forced single thread mode on worker create");
         }
 #else
-        thread_mode = params->thread_mode;
-#endif
-    } else {
-        thread_mode = UCS_THREAD_MODE_SINGLE;
-    }
+        if (params->thread_mode != UCS_THREAD_MODE_SINGLE) {
```

Comment:
minor: maybe do `if (params->thread_mode == UCS_THREAD_MODE_SINGLE) {` and switch if/else?

Reply:
yes, it is better.
but now we pre-init uct_thread_mode `ucs_thread_mode_t uct_thread_mode = UCS_THREAD_MODE_SINGLE;`

Reply:
but pre init should in line 1498
also lets set worker->flags=0 there too, also remote line 1514

Reply:
> but pre init should in line 1498

done

> also lets set worker->flags=0 there too, also remove line 1514

done